src/Plutus/IsoNativeForeign.purs

```diff
@@ -0,0 +1,151 @@
+module Plutus.IsoNativeForeign
+  ( class IsoNativeForeign
+  , toNativeType
+  ) where
+
+import Prelude
+import Data.Array
+import Data.Maybe (Maybe(Just, Nothing))
+import Data.Tuple (Tuple(Tuple), fst)
+import Data.UInt (UInt, fromInt, toInt, (.&.), and, zshr, shl)
+import Data.Newtype (class Newtype, wrap)
+import Types.ByteArray (ByteArray, byteArrayToIntArray, byteArrayFromIntArray)
+import Serialization.Hash (ed25519KeyHashFromBytes, scriptHashFromBytes) as Hash
+import Serialization.Address (addressBytes)
+import Serialization.Address (Address) as Foreign
+import Plutus.Types.Address (Address) as Plutus
+import Plutus.Types.Credential
+  ( Credential(PubKeyCredential, ScriptCredential)
+  , StakingCredential(StakingHash, StakingPtr)
+  )
+
+-- | This type class asserts that two types (native Purescript and
+-- | foreign Javascript) are isomorphic, so they can be converted
+-- | to each other without losing information.
+class IsoNativeForeign :: Type -> Type -> Constraint
```

Comment:
I would perhaps recommend changing the name of the class and adjusting the comment, since the types are not isomorphic given the signature `frgn -> Maybe native`. This means that information loss is indeed possible